keps/0004-add-testing-infrastructure.md

```diff
@@ -19,41 +19,47 @@ superseded-by:
-- [Implementation History](#implementation-history)
-- [Infrastructure Needed [optional]](#infrastructure-needed-optional)
-
-[tools for generating]: https://github.com/ekalinin/github-markdown-toc
+* [Summary](#summary)
+* [Motivation](#motivation)
+   * [Goals](#goals)
+   * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+   * [Tests](#tests)
+      * [Unit tests](#unit-tests)
+      * [Integration tests](#integration-tests)
+      * [Framework tests](#framework-tests)
+   * [Kubernetes clusters](keps/0004-add-testing-infrastructure.md#kubernetes-clusters)
+   * [CICD](#cicd)
+      * [Pull Requests](#pull-requests)
+      * [Master Branch](#master-branch)
+         * [Tests](#tests-1)
+         * [Pushes](#pushes)
+         * [Schedule](#schedule)
+         * [Base Image Change](#base-image-change)
+      * [Tags/Release](#tagsrelease)
+   * [User Stories](#user-stories)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+   * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Infrastructure Needed [optional]](#infrastructure-needed-optional)
 
 ## Summary
 
-As the complexity and scope of KUDO grows, it becomes impossible to manually validate
-exisiting frameworks and capabilities still function as expected. As part of a robust
-CICD pipeline, a testing suite needs to be developed that can validate each commit, pull
-request, or even deployment of KUDO on a cluster.
+As the complexity and scope of KUDO grows, it becomes impossible to manually validate exisiting frameworks and capabilities still function as expected. As part of a robust CICD pipeline, a testing suite needs to be developed that can validate each commit, pull request, or even deployment of KUDO on a cluster.
 
-These testing packages should also assure conformance for the `kudobuilder/frameworks` repository
-to particular versions of KUDO.
+These testing packages should also assure conformance for the `kudobuilder/frameworks` repository to particular versions of KUDO.
+
+[KEP-0008](https://github.com/kudobuilder/kudo/blob/master/keps/0008-framework-testing.md) outlines the design of a testing harness for validating KUDO and frameworks. This document outlines testing procedures and policies - e.g., what, where, when and how we will test.
 
 ## Motivation
 
 ### Goals
 
 - Ensure validation of API objects is functioning correctly
 - Ensure controllers execute known process flows correctly
-- Validate Framework and FrameworkVersions in kudobuilder/frameworks adhere to the API spec defined by Kudo. Provide common
-  misconfigurations and validate that the testing framework notifies the users/developer of failure
+- Validate Framework and FrameworkVersions in kudobuilder/frameworks adhere to the API spec defined by Kudo. Provide common misconfigurations and validate that the testing framework notifies the users/developer of failure
```

Comment:
after 0.3.0 people won't submit FV and F CRDs anymore so I think this is no longer an issue. This will be generated by kudo itself so unit tested... And I think code for verification of the new package format will already be part of CLI so it should be reused for any kind of validation.

Reply:
Removed